agent/agent_pool.go

```diff
@@ -51,18 +51,11 @@ func (r *AgentPool) runWorker(worker *AgentWorker, im *IdleMonitor) error {
 	if err := worker.Connect(); err != nil {
 		return err
 	}
+	// Ensure the worker is disconnected at the end of this function.
+	defer worker.Disconnect();
```

Comment:
I think the `defer worker.Disconnect()` approach is a good trade-off for discarding the error that `worker.Disconnect()` might return, because;

* `agent.AgentWorker.Disconnect()` has [already logged the disconnect error,](https://github.com/buildkite/agent/blob/67c82ef910518f1718f2c65045a8e70691885da3/agent/agent_worker.go#L493-L496)
* it's questionable whether the caller of `runWorker` would want to consider it an error when both Connect & Start were successful, 
* as you mention, `AgentPool.Start` currently only reports one of N errors anyway
    * this is probably a bug, but it reinforces that this error scenario isn't critical